apps/address-service/domains/common/utils/services/search/plugins/SearchByProvider.js

```diff
@@ -0,0 +1,78 @@
+const { AbstractSearchPlugin } = require('@address-service/domains/common/utils/services/search/AbstractSearchPlugin')
+const { Address } = require('@address-service/domains/address/utils/serverSchema')
+const { generateAddressKey } = require('@address-service/domains/common/utils/addressKeyUtils')
+const { getSearchProvider } = require('@address-service/domains/common/utils/services/providerDetectors')
+const get = require('lodash/get')
+
+class SearchByProvider extends AbstractSearchPlugin {
```

Comment:
Why provider extends plugin?

Reply:
This is not a provider. This is a plugin that uses a provider.

Reply:
From name it is not obvious

Reply:
I looked at the naming of keystone plugins while created these ones. Also, all plugins are situated in a separated directory. Also, base class called AbstractSearch**Plugin**